### ISSUE TITLE
Error with blank node syntax/output

### ISSUE THREAD

**Issue by megankatsumi** [ontop/ontop#319]
### Description

I'm running into an error specifying blank nodes as described in https://github.com/ontop/ontop/issues/303#issuecomment-529514018 in my mapping target. While I am able to create blank nodes with the syntax `_:{some column}`, if I attempt to append other values, e.g. "_:birth{some column" to the blank node ontop produces an error. I am running ontop cli 3.0 (have also tried via Protege) on Windows Server2016 standard (64-bit) .

### Steps to Reproduce
For example:
1. A mapping like:
```
mappingId 	MAPID-landcover
target 		<http://esricanada.com/gfx_ontology_prototype/{feature_hash}> a <http://ontology.eil.utoronto.ca/icity/LandUse/Parcel> ; <http://ontology.eil.utoronto.ca/icity/LandUse/hasLandUse> _:{feature_hash} .
source SELECT feature_hash,cover FROM landcover_mun
Will produce triples such as:
<http://esricanada.com/gfx_ontology_prototype/f23c93adffcbee83349174feda7fd7e1> <http://ontology.eil.utoronto.ca/icity/LandUse/hasLandUse> _:b0 .
```
as opposed to what I would expect:
```
<http://esricanada.com/gfx_ontology_prototype/f23c93adffcbee83349174feda7fd7e1> <http://ontology.eil.utoronto.ca/icity/LandUse/hasLandUse> _:f23c93adffcbee83349174feda7fd7e1 .
```
2. If I try to append some other text to the blank node label, e.g.:
```
[MappingDeclaration] @collection [[
mappingId	MAPID-parcel
target		<http://esricanada.com/gfx_ontology_prototype/{feature_hash}> a <http://ontology.eil.utoronto.ca/icity/LandUse/Parcel> ; <http://ontology.eil.utoronto.ca/icity/LandUse/hasLandUse> _:landuse{feature_hash} . 
source		SELECT "feature_hash", "lu_class", "name1" from "landuse_mun"

]]

```
ontop produces the following error:
```
> C:\Program Files\ontop>.\ontop materialize --disable-reasoning -f turtle -m ESRI-GFX-test.obda -o ESRI-output-test-2.ttl -p ESRI-GFX.properties
> java.lang.RuntimeException: it.unibz.inf.ontop.exception.InvalidMappingExceptionWithIndicator:
> The syntax of the mapping is invalid (and therefore cannot be processed). Problems:
> 
> MappingId = 'MAPID-parcel'
> Line 27: Invalid target: '<http://esricanada.com/gfx_ontology_prototype/{feature_hash}> a <http://ontology.eil.utoronto.ca/icity/LandUse/Parcel> ; <http://ontology.eil.utoronto.ca/icity/LandUse/hasLandUse> _:landuse{feature_hash} .'
> Debug information
> Problem parsing in OBDA document.
> Could not load OBDA model. Either a suitable parser could not be found, or parsing failed. See parser logs below for explanation.
> The following parsers were tried:
> 1) TurtleOBDASQLParser
> 
> Details:
> --------------------------------------------------------------------------------
> Parser: TurtleOBDASQLParser
> Syntax error location: column 180
> mismatched input '_:landuse{feature_hash}' expecting {STRING_WITH_CURLY_BRACKET, BOOLEAN_LITERAL, IRIREF_EXT, IRIREF, PREFIXED_NAME, PREFIXED_NAME_EXT, BLANK_NODE_FUNCTION, BLANK_NODE_LABEL, INTEGER, DECIMAL, DOUBLE, INTEGER_POSITIVE, INTEGER_NEGATIVE, DOUBLE_POSITIVE, DOUBLE_NEGATIVE, DECIMAL_POSITIVE, DECIMAL_NEGATIVE, STRING_LITERAL_QUOTE, ANON}
> 
> 
> 
>         at it.unibz.inf.ontop.cli.OntopMaterialize.createMaterializer(OntopMaterialize.java:143)
>         at it.unibz.inf.ontop.cli.OntopMaterialize.run(OntopMaterialize.java:117)
>         at it.unibz.inf.ontop.cli.Ontop.main(Ontop.java:18)
> Caused by: it.unibz.inf.ontop.exception.InvalidMappingExceptionWithIndicator:
> The syntax of the mapping is invalid (and therefore cannot be processed). Problems:
> 
> MappingId = 'MAPID-parcel'
> Line 27: Invalid target: '<http://esricanada.com/gfx_ontology_prototype/{feature_hash}> a <http://ontology.eil.utoronto.ca/icity/LandUse/Parcel> ; <http://ontology.eil.utoronto.ca/icity/LandUse/hasLandUse> _:landuse{feature_hash} .'
> Debug information
> Problem parsing in OBDA document.
> Could not load OBDA model. Either a suitable parser could not be found, or parsing failed. See parser logs below for explanation.
> The following parsers were tried:
> 1) TurtleOBDASQLParser
> 
> Details:
> --------------------------------------------------------------------------------
> Parser: TurtleOBDASQLParser
> Syntax error location: column 180
> mismatched input '_:landuse{feature_hash}' expecting {STRING_WITH_CURLY_BRACKET, BOOLEAN_LITERAL, IRIREF_EXT, IRIREF, PREFIXED_NAME, PREFIXED_NAME_EXT, BLANK_NODE_FUNCTION, BLANK_NODE_LABEL, INTEGER, DECIMAL, DOUBLE, INTEGER_POSITIVE, INTEGER_NEGATIVE, DOUBLE_POSITIVE, DOUBLE_NEGATIVE, DECIMAL_POSITIVE, DECIMAL_NEGATIVE, STRING_LITERAL_QUOTE, ANON}
> 
> 
> 
>         at it.unibz.inf.ontop.spec.mapping.parser.impl.OntopNativeMappingParser.load(OntopNativeMappingParser.java:211)
>         at it.unibz.inf.ontop.spec.mapping.parser.impl.OntopNativeMappingParser.parse(OntopNativeMappingParser.java:115)
>         at it.unibz.inf.ontop.spec.mapping.parser.impl.OntopNativeMappingParser.parse(OntopNativeMappingParser.java:64)
>         at it.unibz.inf.ontop.spec.mapping.impl.AbstractMappingExtractor.extractPPMapping(AbstractMappingExtractor.java:51)
>         at it.unibz.inf.ontop.spec.mapping.impl.AbstractMappingExtractor.extract(AbstractMappingExtractor.java:41)
>         at it.unibz.inf.ontop.spec.impl.DefaultOBDASpecificationExtractor.extract(DefaultOBDASpecificationExtractor.java:38)
>         at it.unibz.inf.ontop.injection.impl.OntopMappingConfigurationImpl.loadSpecification(OntopMappingConfigurationImpl.java:129)
>         at it.unibz.inf.ontop.injection.impl.OntopMappingSQLConfigurationImpl.loadSpecification(OntopMappingSQLConfigurationImpl.java:85)
>         at it.unibz.inf.ontop.injection.impl.OntopMappingSQLAllConfigurationImpl.loadSpecification(OntopMappingSQLAllConfigurationImpl.java:48)
>         at it.unibz.inf.ontop.injection.impl.OntopSQLOWLAPIConfigurationImpl.loadOBDASpecification(OntopSQLOWLAPIConfigurationImpl.java:34)
>         at it.unibz.inf.ontop.injection.impl.OntopOBDAConfigurationImpl.loadSpecification(OntopOBDAConfigurationImpl.java:44)
>         at it.unibz.inf.ontop.materialization.impl.DefaultOntopRDFMaterializer.<init>(DefaultOntopRDFMaterializer.java:94)
>         at it.unibz.inf.ontop.rdf4j.materialization.impl.DefaultRDF4JMaterializer.<init>(DefaultRDF4JMaterializer.java:58)
>         at it.unibz.inf.ontop.rdf4j.materialization.RDF4JMaterializer.defaultMaterializer(RDF4JMaterializer.java:34)
>         at it.unibz.inf.ontop.cli.OntopMaterialize.createMaterializer(OntopMaterialize.java:136)
>         ... 2 more
```

**Comment by juliencorman** [comment]
Hello Megan,

Thank you for the notification.

For the first problem, this is the expected behavior.
According to the RDF standard, a blank node has no identifier.
For instance, the (syntactic) graphs:

:a1  :a2 _:f23c93adffcbee83349174feda7fd7e1 .
:a3  :a4 _:f23c93adffcbee83349174feda7fd7e1 .
:a5  :a6 _:cheesecake .

and 

:a1  :a2 _:b0 .
:a3  :a4 _:b0 .
:a5  :a6 _:b1 .

are semantically the same RDF graph.

So the (syntactic) graph that you expected may not be the one that you query (or materialize) with Ontop. 
Ontop only guarantees that these two graphs are isomorphic and coincide when it comes to proper IRIs (not blank node labels).
And the technical explanation for this difference is that Ontop relies on the RDF4J API, which may alter blank nodes labels.

As for the second issue (the parser exception), thank you very much for notifying it.
This should be fixed by commit 4f52d273.

Regards,
Julien

**Comment by megankatsumi** [comment]
Thanks Julien!
I did notice the comment about the semantics of blank nodes in the documentation, somehow I'd thought that was done simply to reinforce the concept rather than refer to the output of Ontop (was still expecting the values to be encoded as is). Makes sense now though!
